Add Polkadot RPCs to the Dapp examples and wallet

hamidra opened the pull request:
I could not find any CONTRIBUTE docs so hope this is correct.

We are exploring adding wallet-connect support for Polkadot dapps/wallets.
The added Polkadot RPCs in these examples can work as an example implementation for having Wallet-connect support in Polkadot dapps wallets.

These changes:

- Adds Polkadot and its testnet(Westend) as well as RPCs tx and signing to the supported networks in stand-alone dapp example (react-dapp-v2).

- Adds polkadot accounts + message/tx  signing to the wallet

https://youtu.be/faAKiQ6555s

bkrem commented on .gitignore:
```diff
@@ -0,0 +1,107 @@
+# Logs
```
Please remove, each example defines its own `.gitignore` so this shouldn't be needed at the root.

bkrem commented on dapps/react-dapp-v2/src/chains/polkadot.ts:
```diff
@@ -29,15 +51,15 @@ export function getChainMetadata(chainId: string): ChainMetadata {
-          label: "params",
-          value: JSON.stringify(request.params, null, "\t"),
+          label: 'params',
+          value: JSON.stringify(request.params, null, '\t'),
```
Looks like something happened with string formatting in this file specifically? The other diffs still use double quote `""` so not sure what happened 😄

hamidra replied:
It was my prettier which was set to single quote:) fixed.

bkrem commented on dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx:
```diff
@@ -107,6 +113,7 @@ export function JsonRpcContextProvider({
       try {
         setPending(true);
         const result = await rpcRequest(chainId, address);
+        console.log(result);
```
Leftover log

bkrem replied:
```suggestion

```

bkrem commented on dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx:
```diff
@@ -613,14 +620,89 @@ export function JsonRpcContextProvider({
+    testSignTransaction: _createJsonRpcRequestHandler(
+      async (chainId: string, address: string): Promise<IFormattedRpcResponse> => {
+        // Below example is a scale encoded payload for system.remark("this is a test wallet-connect remark") transaction.
+        // decode url: https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.polkadot.io#/extrinsics/decode/0x00019074686973206973206120746573742077616c6c65742d636f6e6e6563742072656d61726b
```
Nice 👍 thank you for context here

bkrem commented on dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx:
```diff
@@ -613,14 +620,89 @@ export function JsonRpcContextProvider({
+              },
+            },
+          });
+          // ToDo: add sig validation after upgrating cra scripts (needs react-scripts ^v5.0.0)
```
Is this referring to react-scripts in the dapp example? 

This example was refactored to next.js a few days ago so might be worth rebasing/merging from `main` if you were implementing against the old CRA setup 👍

hamidra replied:
Right:) I initially made changes on CRA version but after merging  the Next version,  forgot it is not using CRA anymore! It works now and I added back the signature verification.

bkrem commented on dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx:
```diff
@@ -613,14 +620,89 @@ export function JsonRpcContextProvider({
+    ),
+    testSignMessage: _createJsonRpcRequestHandler(
+      async (chainId: string, address: string): Promise<IFormattedRpcResponse> => {
+        // some interfaces like sr25519 need to wait for WASM to load
```
Is this leftover or referring to something specific in this method?

hamidra replied:
The column indeed is misplaced, but it is related to the "await cryptoWaitReady()". In order to use sr25519 signatures in the lib it needs to wait till relevant WASM code loads. I moved the comment to the correct line.

bkrem commented on wallets/react-wallet-v2/src/pages/index.tsx:
```diff
@@ -29,6 +30,9 @@ export default function HomePage() {
         <AccountCard key={name} name={name} logo={logo} rgb={rgb} address={solanaAddress} />
       ))}
+      {Object.values(POLKADOT_MAINNET_CHAINS).map(({ name, logo, rgb }) => (
+        <AccountCard key={name} name={name} logo={logo} rgb={rgb} address={solanaAddress} />
```
```suggestion
        <AccountCard key={name} name={name} logo={logo} rgb={rgb} address={polkadotAddress} />
```

bkrem commented on wallets/react-wallet-v2/src/pages/index.tsx:
```diff
@@ -41,6 +45,9 @@ export default function HomePage() {
             <AccountCard key={name} name={name} logo={logo} rgb={rgb} address={solanaAddress} />
           ))}
+          {Object.values(POLKADOT_TEST_CHAINS).map(({ name, logo, rgb }) => (
+            <AccountCard key={name} name={name} logo={logo} rgb={rgb} address={solanaAddress} />
```
```suggestion
            <AccountCard key={name} name={name} logo={logo} rgb={rgb} address={polkadotAddress} />
```

bkrem reviewed (changes requested):
Great work here @hamidra, thank you for trailblazing Polkadot support in the example dapp and wallet 🙏 

Thank you also for the youtube video showcasing the new functionality 💯 

Some minor fixes required but I think we're good to go once they're resolved

IljaDaderko reviewed (commented):
Looks good to me from wallet example perspective 👍 Thank you for keeping same overall structure as well!

hamidra commented:
Excellent! Thanks. I addressed the comments. 
I'm gonna talk to some polkadot wallet teams to implement the wallet-connect on their side. If any changes happen in the RPCs I will send PRs to keep these up-to-date, but should be good for now.

bkrem commented:
> I'm gonna talk to some polkadot wallet teams to implement the wallet-connect on their side. If any changes happen in the RPCs I will send PRs to keep these up-to-date

Legend 💪

bkrem commented:
Good to go from my side, all good for you @IljaDaderko?

IljaDaderko commented:
@bkrem Yep good to go as well, only concern is failing preview builds for various dapp flavours, but that could be unrelated to this pr?

bkrem commented:
> only concern is failing preview builds for various dapp flavours, but that could be unrelated to this pr?

It's just Vercel gating deployments, so I had to authorize the two relevant ones here (`react-dapp-v2`, `react-wallet-v2`). The other dapp examples need a proper overhaul anyways once we have providers updated for v2 👍
